src/Telnyx.net/Entities/TelnyxCollection.cs

```diff
@@ -0,0 +1,97 @@
+namespace Telnyx.net.Entities
+{
+    using System.Collections;
+    using System.Collections.Generic;
+    using Newtonsoft.Json;
+    using Telnyx.Infrastructure;
+
+    /// <summary>
+    /// Telnyx List.
+    /// </summary>
+    /// <typeparam name="T">Object Type.</typeparam>
+
+    [JsonArray]
+    public class TelnyxCollection<T> : TelnyxEntity, IList<T>
```

Comment:
Why create this new class and not TelynxList? There are a lot of methods on here that aren't implemented explicitly

Reply:
TelnyxList doesn't implement ICollection. TelnyxCollection implements IList which in turn implements ICollection. So, they are serving different purpose. The methods that are not currently used are marked not implemented. Only used methods are implemented.